src/CacheInvalidator.php

```diff
@@ -266,6 +274,24 @@ public function invalidateRegex($path, $contentType = null, $hosts = null)
         return $this;
     }
 
+    /**
+     * Clear the cache completely.
+     *
+     * @throws UnsupportedProxyOperationException if HTTP cache does not support clearing the cache completely
+     *
+     * @return $this
+     */
+    public function clearCache()
+    {
+        if (!$this->cache instanceof ClearCacheCapable) {
```

Comment:
imho this should fallback to BAN everything, if the cache is BanCapable

Reply:
But there is no general method like `banEveryting()` right? I mean, how am I supposed to know if the proxy supports "banning everything"? And if so, how? Does it use a special header with `*` or not?

Reply:
the way ban is defined, if you specify no criteria, it matches everything and thus bans everything.

Reply:
So `->ban()` with no headers speaking in code?

Reply:
genau

Reply:
Tried to implement this but not sure it's a good idea to be honest. 
If we do this I have absolutely no possibility to find out if a proxy really supports clearing or if it just falls back to banning. It might cause issues because I'm never informed about the fact that I'm using a slow fallback cache clearing mechanism.

Imho if someone wants to do that, they should do this:

```php
try {
    $cacheInvalidator->clearCache();
} catch (UnsupportedProxyOperationException $e) {
    $cacheInvalidator->invalidate();
}
```

Reply:
fair point indeed. instead of try-catch, they can also check the capability and call the right method.